components/Promotion/basic_usage.rst

```diff
@@ -0,0 +1,323 @@
+
+.. _component_promotion_checker_promotion-eligibility-checker:
+
+PromotionEligibilityChecker
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This is a service that checks if the promotion rules are eligible for a given subject.
+
+.. code-block:: php
+
+    <?php
+
+    use Sylius\Component\Promotion\Model\Promotion;
+    use Sylius\Component\Promotion\Model\Rule;
+    use Sylius\Component\Promotion\Model\Action;
+    use Sylius\Component\Promotion\Checker\PromotionEligibilityChecker;
+    use Ticket;
+
+    $registry = new ServiceRegistry('Sylius\Component\Promotion\Model\RuleInterface');
+    $dispatcher = new EventDispatcher();
+
+    /**
+     * @param ServiceRegistryInterface $registry
+     * @param EventDispatcherInterface $dispatcher
+     */
+    $checker = new PromotionEligibilityChecker($registry, $dispatcher);
+
+    // Let's create a new promotion
+    $promotion = new Promotion();
+    $promotion->setName('Test');
+
+    // And a new action for that promotion, that will give a fixed discount of 10
+    $action = new Action();
+    $action->setType('fixed_discount');
+    $action->setConfiguration(array('amount' => 10));
+    $action->setPromotion($promotion);
+
+    // That promotion will also have a rule - works for item amounts over 2
+    // You define the behaviour of a rule by setting its type (from the ones already available in the system)
+    $rule = new Rule();
+    $rule->setType('item_count');
+
+    // And next by setting its configuration via an array.
+    $configuration = array('count' => 2);
+    $rule->setConfiguration($configuration);
+
+    // The system will know about a rule pnly if you register it before using the ServiceRegistry.
+    // You should register the rule using types already defined in your system
+    $registry->register('item_count', $rule);
+
+    // Next connect the rule to your promotion.
+    $promotion->addRule($rule);
+
+    // Now we need an object that implements the PromotionSubjectInterface
+    // so use your custom Ticket class.
+    $subject = new Ticket();
+
+    $subject->addPromotion($promotion);
+    $subject->setQuantity(3);
+    $subject->setUnitPrice(10);
+
+    $checker->isEligible($subject, $promotion); // Returns true
+
+.. note::
+
+    It implements the :ref:`component_promotion_checker_promotion-eligibility-checker-interface`.
+
+.. _component_promotion_action_promotion-applicator:
+
+PromotionApplicator
+-------------------
+
+In order to automate the process of promotion application the component provides us with a Promotion Applicator,
+which is able to apply and revert single promotions on a subject implementing the **PromotionSubjectInterface**.
+
+.. code-block:: php
+
+    <?php
+
+    use Sylius\Component\Promotion\Action\PromotionApplicator;
+    use Sylius\Component\Promotion\Model\Promotion;
+    use Sylius\Component\Registry\ServiceRegistry;
+    use Ticket;
+
+    $registry = new ServiceRegistry('Sylius\Component\Promotion\Model\ActionInterface');
```

Comment:
Let's maybe mention here that you need to add your own ActionsInterface implementation, hm? I mean - add action to registry.